Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -229,17 +233,19 @@ public final class JSONRPCConnection {
               continue MESSAGE_LOOP
             }
           case .unknown:
-            log("error decoding message: \(error.message)", level: .error)
+            send(.errorResponse(ResponseError(error), id: nil),
```

Comment:
We can't test this behaviour directly due to the fatalError, but it might be good to add a test for the JSONRPCConnection receiving an error with `nil` id.  For example, if you manually send an `errorResponse(..., id: nil)` make sure it doesn't crash the receiving connection.  This would fit in `Tests/LanguageServerProtocolJSONRPCTests/ConnectionTests.swift`

Reply:
With this one I better ask you before going down the wrong rabbit hole ... I don't see an easy way for testing that a `JSONRPCMessage`s or a concrete error response reaches the receiver. I'd want to write somethin like this:

```
func testSendErrorResponseWithNilID() {
    let client = connection.client

    let expectation = self.expectation(description: "received message")
    client.handleNextMessage { (message: JSONRPCMessage) in
      expectation.fulfill()
    }
  
    connection.serverConnection._send(.errorResponse(.cancelled, id: nil))

    waitForExpectations(timeout: 10)
  }
```

Reply:
Ah I see we don't have a good way to test this right now.  Let's skip this.